### PR TITLE
WebIDL: use Uint8Array

### LINKED ISSUE #31064: WebIDL: use TypedArray
As as follow-up to https://github.com/servo/servo/pull/30990, we should do the same for other typed arrays. 

The list is found at https://github.com/servo/servo/blob/92196d985dceb0ca708b097e2a847b255d8387c8/third_party/WebIDL/WebIDL.py#L3634

See the various arrays which currently default to `IDLType.Tags.interface`, and which currently all fallback to using a `JSObject` in [codegen](https://github.com/servo/servo/blob/340a86c37584079e6387fe7dec8b8bbf3c722411/components/script/dom/bindings/codegen/CodegenRust.py). Each should be changed to use it's own type, like was done in the linked-to PR for `FLoat32Array`. Note that not all may be currently used by Servo, to find out if they are, search for it in `*.webidl` files. 

Each could be done independently, although it might make sense to investigate making [`components/script/dom/bindings/typedarrays.rs`](https://github.com/servo/servo/blob/340a86c37584079e6387fe7dec8b8bbf3c722411/components/script/dom/bindings/typedarrays.rs) generic first. 

If the typed array is currently stored in a `Heap`, then a wrapper like [`HeapFloat32Array`](https://github.com/servo/servo/pull/30990/files#diff-8f10f5d3caee3cce7befe080a3333a5f9402a651884287734f41e786e4323181R30) would be useful, otherwise the unsafe code can simply be hidden behind utilities, such as [`create_float32_array`](https://github.com/servo/servo/pull/30990/files#diff-8f10f5d3caee3cce7befe080a3333a5f9402a651884287734f41e786e4323181R16)

The actual typed array JS types are found at https://github.com/servo/mozjs/blob/7184f658b7e2de8a8d288ff39001364513115782/mozjs/src/typedarray.rs

part of https://github.com/servo/servo/issues/30862

### PR DESCRIPTION
**Pull request by Taym95**
Part of #31064

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix 

- [x] These changes do not require tests because there are no behavior changes.

### CODE REVIEW

File: components/script/dom/bindings/typedarrays.rs [hunk 1]
```diff
@@ -112,3 +102,21 @@ impl HeapFloat32Array {
         }
     }
 }
+
+pub fn create_typed_array<T, S>(
```
Comment (Taym95) [inline components/script/dom/bindings/typedarrays.rs:106]: @gterzian can you please review this one?

**Review by gterzian — APPROVED** [review]
Great work, thanks! LGTM, with just one nit. Leaving the issue open because there are still other typed arrays to use(right?).

**Comment by Taym95** [comment]
> Great work, thanks! LGTM, with just one nit. Leaving the issue open because there are still other typed arrays to use(right?).

Yes, and I will make other PR to address other typed arrays

File: components/layout_2020/flow/inline.rs [hunk 2]
```diff
@@ -2405,6 +2405,7 @@ impl<'a> ContentSizesComputation<'a> {
             had_non_whitespace_content_yet: false,
             linebreaker: None,
             ending_inline_pbm_stack: Vec::new(),
-        }.traverse(inline_formatting_context)
+        }
+        .traverse(inline_formatting_context)
```
[the remark this answers is not in the document]
Comment (Taym95) [reply components/layout_2020/flow/inline.rs]: Yes I will revert, that was result of `fmt`

**Pull request merged**